server/src/main/java/io/druid/metadata/DerbyMetadataStorageActionHandler.java

```diff
@@ -46,34 +46,45 @@
 
   @Override
   protected Query<Map<String, Object>> createInactiveStatusesSinceQuery(
-      Handle handle, DateTime timestamp, @Nullable Integer maxNumStatuses
+      Handle handle, DateTime timestamp, @Nullable Integer maxNumStatuses, @Nullable String datasource
   )
   {
+    final String whereStmt = getWhereClause(datasource);
     String sql = StringUtils.format(
         "SELECT "
         + "  id, "
         + "  status_payload, "
-        + " created_date, "
-        + " datasource, "
-        + " payload "
+        + "  created_date, "
+        + "  datasource, "
+        + "  payload "
         + "FROM "
         + "  %s "
         + "WHERE "
-        + "  active = FALSE AND created_date >= :start "
+        + whereStmt
```

Comment:
nit: `whereStmt` can be replaced with `geWhereClause(datasource)`, we can get rid of the extra variable there

Reply:
done